tests/connectors/opsani_dev_test.py

```diff
@@ -1194,3 +1274,28 @@ async def _remedy_check(id: str, *, config, deployment, kube_port_forward, load_
 
     else:
         raise AssertionError(f"unhandled check: '{id}'")
+
+async def _run_remedy_from_check(failure: servo.Check) -> None:
+    """Replicate the application remedies as done in ServoCLI.check_servo with the remedy argument set to True"""
+    assert failure.remedy, f"Expected check to have remedy method: {failure}"
+
+    if asyncio.iscoroutinefunction(failure.remedy):
+        task = asyncio.create_task(failure.remedy())
```

Comment:
this line is different, which is the right form (may not matter)  with failure.remedy() or failure.remedy

Reply:
The right form depends on the state of remedy; whether or not the async function was called before being passed into the check.

If it was not called before/during check instantiation, then it is a coroutine function that would need to be called before being scheduled with create_task. `asyncio.iscoroutinefunction` returns true in this case and remedy is called with `failure.remedy()`.

If remedy was called before/during the check instantiation then it is already coroutine object that simply needs to be scheduled with create_task. In this case, `asyncio.iscoroutinefunction` returns false and `asyncio.iscoroutine` returns true. The remedy is scheduled (`asyncio.create_task(failure.remedy)`) without calling it since it has already been called

Reply:
Thanks for the clarification!  LGTM